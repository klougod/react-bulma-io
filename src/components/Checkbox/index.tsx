import React, { InputHTMLAttributes, ReactNode } from 'react'

interface CheckboxProps extends InputHTMLAttributes<HTMLInputElement> {
  checked?: boolean
  onChange?: () => void
  text?: ReactNode
  wrapperClass?: string
}

export const Checkbox = ({
  text,
  checked,
  className,
  wrapperClass,
  onChange,
  ...rest
}: CheckboxProps) => (
  <div className={`checkbox ${wrapperClass}`}>
    <input
      type='checkbox'
      id='bulma-io-checkbox'
      className={className}
      checked={checked}
      onChange={onChange}
      {...rest}
    />
    <label htmlFor='bulma-io-checkbox' onChange={onChange}>
      &nbsp;{text}
    </label>
  </div>
)

Checkbox.defaultProps = {
  checked: false,
  onChange: null,
  text: '',
  className: '',
  wrapperClass: ''
}

export default Checkbox
