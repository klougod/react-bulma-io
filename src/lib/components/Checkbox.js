import React from 'react'
import PropTypes from 'prop-types'

export const Checkbox = ({ text, checked, name, className, wrapperClass, onChange }) => (
  <div className={`checkbox ${wrapperClass}`}>
    <input type="checkbox" className={className} checked={checked} onChange={onChange} name={name} />
    <label>&nbsp;{ text }</label>
  </div>
)

Checkbox.defaultProps = {
  checked: false,
  onChange: null,
  name: '',
  text: '',
  className: '',
  wrapperClass: ''
}

Checkbox.propTypes = {
  checked: PropTypes.bool,
  onChange: PropTypes.func,
  name: PropTypes.string,
  text: PropTypes.string,
  className: PropTypes.string,
  wrapperClass: PropTypes.string
}

export default Checkbox;