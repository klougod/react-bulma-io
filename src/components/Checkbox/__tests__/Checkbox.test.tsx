import React from 'react'
import Enzyme from 'enzyme'

import Checkbox from '../'

describe('Checkbox component test', function () {
  it('should render without crash without props', function () {
    const checkboxMock = Enzyme.shallow(<Checkbox />)
    expect(checkboxMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const checkboxMockProps = {
      className: 'checkboxCLass',
      name: 'checkboxName'
    }
    const checkboxMock = Enzyme.shallow(<Checkbox {...checkboxMockProps} />)
    expect(checkboxMock).toMatchSnapshot()
  })

  it('should call onchange function prop if input changes', function () {
    const checkboxMockProps = { onChange: jest.fn() }
    const checkboxMock = Enzyme.shallow(<Checkbox {...checkboxMockProps} />)
    expect(
      checkboxMock
        .find('input')
        .at(0)
        .prop('onChange')
    ).toHaveBeenCalledTimes(0)
    checkboxMock
      .find('input')
      .at(0)
      .simulate('change', { target: {} })
    expect(
      checkboxMock
        .find('input')
        .at(0)
        .prop('onChange')
    ).toHaveBeenCalledTimes(1)
  })
})
