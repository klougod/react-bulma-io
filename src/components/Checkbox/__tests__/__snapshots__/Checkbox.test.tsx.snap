// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component test should render without crash with props 1`] = `
<div
  className="checkbox "
>
  <input
    checked={false}
    className="checkboxCLass"
    id="bulma-io-checkbox"
    name="checkboxName"
    onChange={null}
    type="checkbox"
  />
  <label
    htmlFor="bulma-io-checkbox"
    onChange={null}
  >
     
  </label>
</div>
`;

exports[`Checkbox component test should render without crash without props 1`] = `
<div
  className="checkbox "
>
  <input
    checked={false}
    className=""
    id="bulma-io-checkbox"
    name=""
    onChange={null}
    type="checkbox"
  />
  <label
    htmlFor="bulma-io-checkbox"
    onChange={null}
  >
     
  </label>
</div>
`;
